controllers/etcd_controller.go

```diff
@@ -791,146 +804,167 @@ func (r *EtcdReconciler) getStatefulSetFromEtcd(etcd *druidv1alpha1.Etcd, values
 	return decoded, nil
 }
 
-func (r *EtcdReconciler) reconcileCronJob(ctx context.Context, logger logr.Logger, etcd *druidv1alpha1.Etcd, values map[string]interface{}) (*batchv1.CronJob, error) {
-	logger.Info("Reconcile etcd compaction cronjob")
-
-	var cronJob batchv1.CronJob
-	err := r.Get(ctx, types.NamespacedName{Name: getCronJobName(etcd), Namespace: etcd.Namespace}, &cronJob)
-
-	//If backupCompactionSchedule is present in the etcd spec, continue with reconciliation of cronjob
-	//If backupCompactionSchedule is not present in the etcd spec, do not proceed with cronjob
-	//reconciliation. Furthermore, delete any already existing cronjobs corresponding with this etcd
-	backupCompactionScheduleFound := false
-	if values["backup"].(map[string]interface{})["backupCompactionSchedule"] != nil {
-		backupCompactionScheduleFound = true
-	}
-
-	if !backupCompactionScheduleFound {
-		if err == nil {
-			err = r.Delete(ctx, &cronJob, client.PropagationPolicy(metav1.DeletePropagationForeground))
-			if err != nil {
-				return nil, err
+func (r *EtcdReconciler) reconcileFullLease(ctx context.Context, logger logr.Logger, etcd *druidv1alpha1.Etcd) (*coordinationv1.Lease, error) {
+	// Get or Create full_snapshot_revisions lease object that will help to set BackupReady condition
+	nsName := types.NamespacedName{
+		Name:      getFullLease(etcd),
+		Namespace: etcd.Namespace,
+	}
+
+	fullLease := &coordinationv1.Lease{}
+	err1 := r.Get(ctx, nsName, fullLease)
+	if err1 != nil {
+		logger.Info("Couldn't fetch full snap lease full-snapshot-revisions because: " + err1.Error())
```

Comment:
Replace `full-snapshot-revisions` with actual full snap name here and other occurrences as well